### PR TITLE
feat: 🎸 methods to unban single user and batch-unban multiple users (no interface yet) 

### LINKED ISSUE #1797: Create batch-unban and single unban endpoints
Routes for unbanning a single user or multiple users at once should be created and extensively tested.

### PR DESCRIPTION
**Pull request by PeculiarE**
Fixes #1797

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

### CODE REVIEW

File: app/controllers/spam_controller.rb [hunk 1]
```diff
@@ -61,7 +66,7 @@ def publish_map
     @map = Map.find(params[:id])
     if check_and_publish(@map)
       notice_text = 'Map published.'
-      notice_text.chop! << ' and author unbanned.' if check_and_unban(@map)
+      notice_text.chop! << ' and author unbanned.' if check_and_unban(@map, 'map')
```
Comment (cesswairimu) [inline app/controllers/spam_controller.rb:69]: cool :heart:

File: test/controllers/spam_controller_test.rb [hunk 2]
```diff
@@ -414,4 +418,77 @@ def users_custom_setup
     assert_equal '0 authors banned.', flash[:notice]
     assert_redirected_to root_path
   end
+
```
Comment (cesswairimu) [inline test/controllers/spam_controller_test.rb:421]: Love the tests!

**Review by cesswairimu — APPROVED** [review]
This is looking good @PeculiarE, good job :tada: :tada: .
Maybe you could do some refactoring on a follow up Pr for some of the things code climate is highlighting?  These could also be good ftos am happy to help on those. Really love the tests! thanks

File: test/controllers/spam_controller_test.rb (hunk 2 above)
Comment (PeculiarE) [reply test/controllers/spam_controller_test.rb:421]: Thank you, Cess ❤️

File: test/controllers/spam_controller_test.rb [hunk 3]
```diff
@@ -10,15 +10,20 @@ def setup #called before every single test
...
+      @maps.uniq.collect { |map|
         map.spam
         map.user.ban
       }
     end
     @map_ids = @maps.collect(&:id).join(',')
   end
 
-  def users_custom_setup
+  def users_custom_setup(ban_users)
```
Comment (jywarren) [inline test/controllers/spam_controller_test.rb:21]: Here, using a boolean as the attribute can make it a little unclear how it's being used lower down. I know this is just a test, but could we as a good habit try:

```suggestion
  def users_custom_setup(ban_users: true) # setting default true?
```

then the usage would be:

```ruby
users_custom_setup(ban_users: false)
```

It's a little more verbose but much clearer where you're using the method lower down. What do you think?

**Comment by jywarren** [comment]
This looks great, i just made one request on named attributes in a test method. Then we can merge this!

File: test/controllers/spam_controller_test.rb (hunk 3 above)
Comment (PeculiarE) [reply test/controllers/spam_controller_test.rb]: I think this makes perfect sense @jywarren, thank you. I've implemented the changes. It does look better now 😄

**Comment by jywarren** [comment]
Woohoo! Great work.

**Pull request merged**